x-pack/plugins/elastic_assistant/server/lib/langchain/execute_custom_llm_chain/index.ts

```diff
@@ -59,25 +67,25 @@ export const callAgentExecutor = async ({
   );
 
   const modelExists = await esStore.isModelInstalled();
-  if (!modelExists) {
-    throw new Error(
-      'Please ensure ELSER is configured to use the Knowledge Base, otherwise disable the Knowledge Base in Advanced Settings to continue.'
-    );
-  }
 
   // Create a chain that uses the ELSER backed ElasticsearchStore, override k=10 for esql query generation for now
   const chain = RetrievalQAChain.fromLLM(llm, esStore.asRetriever(10));
 
-  // TODO: Dependency inject these tools
-  const tools: Tool[] = [
-    new ChainTool({
-      name: 'ESQLKnowledgeBaseTool',
-      description:
-        'Call this for knowledge on how to build an ESQL query, or answer questions about the ES|QL query language.',
-      chain,
-      tags: ['esql', 'query-generation', 'knowledge-base'],
-    }),
-  ];
+  const tools: Tool[] = getApplicableTools({
+    allow,
+    allowReplacement,
+    alertsIndexPattern,
+    assistantLangChain,
+    chain,
+    esClient,
+    modelExists,
+    onNewReplacements,
+    replacements,
+    request,
```

Comment:
nit: as discussed when pairing, at some point we'll scrape the dependencies off the request and use those instead (perhaps I'll do this when wrapping up the tool registration work :).